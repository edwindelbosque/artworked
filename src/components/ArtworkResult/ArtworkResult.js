import React, { Component } from 'react';
import './ArtworkResult.scss';
import loadingGif from '../../assets/loadingGif.gif';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

class Artwork extends Component {
	constructor() {
		super();
		this.state = {
			isLoaded: false
		};
	}

	loadHandler = () => {
		this.setState({
			isLoaded: true
		});
	};

	render() {
		const { isLoaded } = this.state;
		const { name, hqArtwork, id, isFavorites } = this.props;
		const routeHandler = isFavorites ? 'favorites' : 'search';
		const show = isLoaded ? { display: 'block' } : { display: 'none' };
		return (
			<article className='ArtworkResult'>
				{!this.state.isLoaded && <img alt='loading gif' src={loadingGif} />}
				<Link to={`/${routeHandler}/${id}`}>
					<img
						alt={`${name} album artwork`}
						src={hqArtwork}
						style={show}
						onLoad={this.loadHandler}
					/>
				</Link>
			</article>
		);
	}
}

const mapStateToProps = ({ isFavorites }) => ({
	isFavorites
});

export default connect(mapStateToProps)(Artwork);
