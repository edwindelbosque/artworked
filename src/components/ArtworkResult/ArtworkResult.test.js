import React from 'react';
import { ArtworkResult } from './ArtworkResult';
import { shallow } from 'enzyme';

describe('ArtworkResult', () => {
	let wrapper;
	const result = { information: 'yes' };
	const match = { params: ['path'] };

	beforeEach(() => {
		wrapper = shallow(<ArtworkResult />);
	});

	it('should match App snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
