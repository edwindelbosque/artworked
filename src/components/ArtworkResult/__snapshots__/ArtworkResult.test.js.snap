// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtworkResult should match App snapshot 1`] = `
<article
  className="ArtworkResult"
>
  <img
    alt="loading gif"
    src="loadingGif.gif"
  />
  <Link
    to="/search/undefined"
  >
    <img
      alt="undefined album artwork"
      onLoad={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    />
  </Link>
</article>
`;
